From fcc785ecb353e0a8ca19b6745d09df4b5c297987 Mon Sep 17 00:00:00 2001
From: Satheesh Paul <psatheesh@marvell.com>
Date: Wed, 13 Dec 2023 13:48:14 +0530
Subject: [PATCH 748/955] net/cnxk: support PPPoE flow item type

Support to parse RTE_FLOW_ITEM_TYPE_PPPOES rte_flow item
type for cnxk device.

Signed-off-by: Satheesh Paul <psatheesh@marvell.com>
Change-Id: Icd326654743a1cc7d1224aa396120ba920438d97
Reviewed-by: Kiran Kumar K <kirankumark@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/dataplane/dpdk/+/118293
Reviewed-by: Kiran Kumar Kokkilagadda <kirankumark@marvell.com>
Base-Builds: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
Base-Tests: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
Tested-by: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
---
 doc/guides/nics/features/cnxk.ini     | 1 +
 doc/guides/nics/features/cnxk_vec.ini | 1 +
 doc/guides/nics/features/cnxk_vf.ini  | 1 +
 drivers/net/cnxk/cnxk_flow.c          | 7 ++++---
 4 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/doc/guides/nics/features/cnxk.ini b/doc/guides/nics/features/cnxk.ini
index 0451ddd910b2f..806f0b85b728b 100644
--- a/doc/guides/nics/features/cnxk.ini
+++ b/doc/guides/nics/features/cnxk.ini
@@ -70,6 +70,7 @@ ipv6_frag_ext        = Y
 mark                 = Y
 mpls                 = Y
 nvgre                = Y
+pppoes               = Y
 raw                  = Y
 represented_port     = Y
 sctp                 = Y
diff --git a/doc/guides/nics/features/cnxk_vec.ini b/doc/guides/nics/features/cnxk_vec.ini
index 5d0976e6ceadc..083ec19ae0226 100644
--- a/doc/guides/nics/features/cnxk_vec.ini
+++ b/doc/guides/nics/features/cnxk_vec.ini
@@ -65,6 +65,7 @@ ipv6_frag_ext        = Y
 mark                 = Y
 mpls                 = Y
 nvgre                = Y
+pppoes               = Y
 raw                  = Y
 sctp                 = Y
 tcp                  = Y
diff --git a/doc/guides/nics/features/cnxk_vf.ini b/doc/guides/nics/features/cnxk_vf.ini
index e7688a8ae0381..7a5f327632dce 100644
--- a/doc/guides/nics/features/cnxk_vf.ini
+++ b/doc/guides/nics/features/cnxk_vf.ini
@@ -61,6 +61,7 @@ ipv6_frag_ext        = Y
 mark                 = Y
 mpls                 = Y
 nvgre                = Y
+pppoes               = Y
 raw                  = Y
 represented_port     = Y
 sctp                 = Y
diff --git a/drivers/net/cnxk/cnxk_flow.c b/drivers/net/cnxk/cnxk_flow.c
index bf51129600beb..6a78b5a99f7f3 100644
--- a/drivers/net/cnxk/cnxk_flow.c
+++ b/drivers/net/cnxk/cnxk_flow.c
@@ -40,11 +40,12 @@ const struct cnxk_rte_flow_term_info term[] = {
 	[RTE_FLOW_ITEM_TYPE_TX_QUEUE] = {ROC_NPC_ITEM_TYPE_TX_QUEUE,
 					 sizeof(struct rte_flow_item_tx_queue)},
 	[RTE_FLOW_ITEM_TYPE_REPRESENTED_PORT] = {ROC_NPC_ITEM_TYPE_REPRESENTED_PORT,
-						 sizeof(struct rte_flow_item_ethdev)}};
+						 sizeof(struct rte_flow_item_ethdev)},
+	[RTE_FLOW_ITEM_TYPE_PPPOES] = {ROC_NPC_ITEM_TYPE_PPPOES,
+				       sizeof(struct rte_flow_item_pppoe)}};
 
 static int
-npc_rss_action_validate(struct rte_eth_dev *eth_dev,
-			const struct rte_flow_attr *attr,
+npc_rss_action_validate(struct rte_eth_dev *eth_dev, const struct rte_flow_attr *attr,
 			const struct rte_flow_action *act)
 {
 	const struct rte_flow_action_rss *rss;
-- 
2.25.1

